### PR TITLE
Feature/fixes

### PR DESCRIPTION
**Pull request by 0xvoider42**

### CODE REVIEW

File: config.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
```
Comment (Carel155) [inline config.ts]: Probably a better name for the file would be `api.ts`. Or even better would move this to `src/services/api/index.ts`

File: src/components/NewTodo/NewTodo.tsx [hunk 2]
```diff
@@ -1,47 +1,53 @@
...
+    });
 
-    const todoData = {
-      title: enteredTitle,
-      text: enteredText,
-    };
+    return response;
+  };
+
+  const { mutate } = useMutation(addTodoHandler);
```
Comment (Carel155) [inline src/components/NewTodo/NewTodo.tsx]: Move this to top of the component

File: config.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+export const instance = axios.create({
```
Comment (Carel155) [inline config.ts]: Also, give it a better name than instance. Maybe something like `api`

File: src/components/NewTodo/NewTodo.tsx [hunk 4]
```diff
@@ -1,47 +1,53 @@
...
 
-    onAddTodo(todoData);
+  const submitHandler = (data: Todo) => {
+    mutate(data);
   };
 
   return (
-    <form onSubmit={submitHandler}>
+    <form onSubmit={handleSubmit(submitHandler)}>
```
Comment (Carel155) [inline src/components/NewTodo/NewTodo.tsx:20]: No need to use the `submitHandler` also. Just use the `mutate` function

File: src/components/UI/InputField.tsx [hunk 5]
```diff
@@ -1,59 +1,64 @@
...
-    event.preventDefault();
+const UpdateTodo = () => {
+  const { register, handleSubmit } = useForm();
 
-    const enteredId = idRef.current!.value;
-    const enteredTitle = titleRef.current.value;
-    const enteredText = textRef.current.value;
+  const updateTodo = (updateTodo: Todo) => {
+    const { id } = updateTodo;
+    const { title, text } = updateTodo;
```
Comment (Carel155) [inline src/components/UI/InputField.tsx]: You can combine there 2 lines

File: src/components/NewTodo/NewTodo.tsx (hunk 4 above)
Comment (0xvoider42) [reply src/components/NewTodo/NewTodo.tsx:20]: couldn't get it working with only mutate

File: src/components/UI/InputField.tsx [hunk 6]
```diff
@@ -1,59 +1,52 @@
...
 
-  const submitHandler = (event) => {
-    event.preventDefault();
+const UpdateTodo = () => {
+  const { mutate } = useMutation(editTodo);
 
-    const enteredId = idRef.current!.value;
-    const enteredTitle = titleRef.current.value;
-    const enteredText = textRef.current.value;
+  const { register, handleSubmit } = useForm();
```
Comment (Carel155) [inline src/components/UI/InputField.tsx]: Maybe add some `defaultValues`?

File: src/components/UI/ThemeSwitcher.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  }, []);
+
+  if (!mounted) return null;
+
+  return (
+    <Button
+      variant="outlined"
+      onClick={() => {
+        if (mode === "light") {
+          setMode("dark");
```
Comment (Carel155) [inline src/components/UI/ThemeSwitcher.tsx]: Easier would be `setMode(mode === "light" ? "dark" : "light")`

File: src/components/UI/TodoTable.tsx [hunk 8]
```diff
@@ -6,20 +6,29 @@ import {
...
+const TodoTable: ({ todos }: { todos: Todo[] }) => JSX.Element = ({
+  todos,
+}) => {
+  const { mutate } = useMutation(deleteTodo);
+
+  const fetchTodo = () => {
+    return todos;
+  };
+
+  const { data, isLoading, isError } = useQuery("todos", fetchTodo);
```
Comment (Carel155) [inline src/components/UI/TodoTable.tsx]: This seems very weird. We are always just returning todos that are passed in as props? What's the point of using `useQuery` in here?

File: src/components/UI/TodoTable.tsx [hunk 9]
```diff
@@ -6,20 +6,29 @@ import {
...
+
+  if (isLoading) return <LinearProgress color="secondary" />;
+
+  if (isError) return <Alert severity="error">Something went wrong</Alert>;
```
Comment (Carel155) [inline src/components/UI/TodoTable.tsx]: I assume these states are never triggered as we are not loading them really, just returning the props todos. Also don't do inline return

File: src/models/todo.ts [hunk 10]
```diff
@@ -1,9 +1,9 @@
 interface Todo {
```
Comment (Carel155) [inline src/models/todo.ts]: You should probably separate types, one that comes from API always has id, createdAt and updatedAt. The ones that you create in don't have those.

File: src/pages/index.tsx [hunk 11]
```diff
@@ -1,39 +1,26 @@
...
-
-    return response;
-  };
-
-  const removeTodoHandler = (id: number) => {
-    return removeTodo(id);
-  };
-
-  return <TodoTable todos={todos} onRemoveTodo={removeTodoHandler} />;
+// todos is not accessible in Props, should I create a separate type/interface for this case?
```
Comment (Carel155) [inline src/pages/index.tsx]: Probably only add Props only to here, not in the `getServerSideProps`

File: src/services/queries/add-todo.ts [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import { AxiosResponse } from "axios";
+import Todo from "../../models/todo";
+import { api } from "../api";
+
+export const addTodo = (todoData: Todo): Promise<AxiosResponse<Todo>> => {
+  const response = api.post("/todos", {
```
Comment (Carel155) [inline src/services/queries/add-todo.ts]: Just do

```
return api.post("/todos", {
    title: todoData.title,
    text: todoData.text,
  });
```

File: src/services/queries/delete-todo.ts [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import { AxiosResponse } from "axios";
+import { api } from "../api";
+
+export const deleteTodo = (id: number): Promise<AxiosResponse> => {
+  const response = api.delete(`/todos/${id}`);
```
Comment (Carel155) [inline src/services/queries/delete-todo.ts]: Same as above

File: src/services/queries/edit-todo.ts [hunk 14]
```diff
@@ -0,0 +1,10 @@
+import { AxiosResponse } from "axios";
+import Todo from "../../models/todo";
+import { api } from "../api";
+
+export const editTodo = (updateTodo: Todo): Promise<AxiosResponse<Todo>> => {
+  const { id, title, text } = updateTodo;
+  const response = api.patch(`/todos/${id}`, { title, text });
```
Comment (Carel155) [inline src/services/queries/edit-todo.ts]: Same as above

File: src/services/queries/get-todos.ts [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import { api } from "../api";
+
+export const getTodos = () => {
+  const response = api.get("/todos").then((res) => {
```
Comment (Carel155) [inline src/services/queries/get-todos.ts]: Same as above

File: src/components/UI/TodoTable.tsx (hunk 8 above)
Comment (0xvoider42) [reply src/components/UI/TodoTable.tsx]: was just a practice run

File: src/components/UI/TodoTable.tsx (hunk 9 above)
Comment (0xvoider42) [reply src/components/UI/TodoTable.tsx]: loading is hit for a sec, but it's not necessary

File: src/models/todo.ts [hunk 16]
```diff
@@ -1,9 +1,13 @@
-interface Todo {
-  id?: number;
+export interface receiveTodo {
```
Comment (Carel155) [inline src/models/todo.ts:1]: Better name would be `ApiTodo`

File: src/pages/index.tsx [hunk 17]
```diff
@@ -19,7 +17,7 @@ export const getServerSideProps: () => Promise<Props> = async () => {
 };
 
 // todos is not accessible in Props, should I create a separate type/interface for this case?
```
Comment (Carel155) [inline src/pages/index.tsx]: You can remove this comment now

File: src/components/UI/InputField.tsx [hunk 18]
```diff
@@ -2,13 +2,15 @@ import { Button, Grid, Stack, TextField } from "@mui/material";
...
-import Todo from "../../models/todo";
+import { Todo } from "../../models/todo";
 import { editTodo } from "../../services/queries/edit-todo";
 
 const UpdateTodo = () => {
   const { mutate } = useMutation(editTodo);
 
-  const { register, handleSubmit } = useForm();
+  const { register, handleSubmit } = useForm({
+    defaultValues: { id: "id", title: "title", text: "text" },
```
Comment (Carel155) [inline src/components/UI/InputField.tsx]: By `defaultValues` i was thinking just empty string on null

**Pull request merged**